src/services/refactors/moveToNewFile.ts

```diff
@@ -379,7 +389,7 @@ function updateNamespaceLikeImport(
         for (const ref of toChange) {
             changes.replaceNode(sourceFile, ref, factory.createIdentifier(newNamespaceName));
         }
-        changes.insertNodeAfter(sourceFile, oldImportNode, updateNamespaceLikeImportNode(oldImportNode, newModuleName, newModuleSpecifier));
+        changes.insertNodeAfter(sourceFile, oldImportNode, updateNamespaceLikeImportNode(oldImportNode, preferredNewNamespaceName, newModuleSpecifier));

```

Comment:
This line is the bug fix—everything else was cleanup. `newModuleName` wasn’t guaranteed to be a valid identifier.